packages/app/src/contacts/components/contacts/contacts.component.tsx

```diff
@@ -272,8 +272,52 @@ const Contacts: FunctionComponent<ContactsProps> = ({
           true
         )
 
-        const { payload } = await delayResponse(editContact(contact))
+        const { message, payload } =
+          (await delayResponse(addNewContact(contact))).payload ?? {}
+
+        if (payload || message) {
+          const newError: FormError[] = []
+          if (
+            message === "phone-number-duplicated" &&
+            payload?.primaryPhoneNumberIsDuplicated
+          ) {
+            newError.push({
+              field: "primaryPhoneNumber",
+              error: "component.formErrorNumberUnique",
+            })
+          }
+          if (
+            message === "phone-number-duplicated" &&
+            payload?.secondaryPhoneNumberIsDuplicated
+          ) {
+            newError.push({
+              field: "secondaryPhoneNumber",
+              error: "component.formErrorNumberUnique",
+            })
+          }
+          if (message === "Create contact: Empty primary phone number") {
+            newError.push({
+              field: "primaryPhoneNumber",
+              error: "component.formErrorRequiredPrimaryPhone",
+            })
+          }
+          if (newError.length === 0) {
+            newError.push({
+              field: "primaryPhoneNumber",
+              error: "component.formErrorNumberUnique",
+            })
+          }
```

Comment:
I can see two things :D 
- those messages - "phone-number-duplicated" and "Create contact: Empty primary phone number" could be in enum/const
- if's that have comparison to message could be chained, that way u won't compare it twice :)